src/gui/folderwatcher.cpp

```diff
@@ -45,6 +49,25 @@ FolderWatcher::FolderWatcher(Folder *folder)
     : QObject(folder)
     , _folder(folder)
 {
+    _timer.setInterval(notificationTimeoutC);
+    connect(&_timer, &QTimer::timeout, this, [this] {
+        auto paths = std::move(_changeSet);
+        // ------- handle ignores:
+        for (const auto &path : paths) {
+            // we cause a file change from time to time to check whether the folder watcher works as expected
+            if (!_testNotificationPath.isEmpty()
+                && Utility::fileNamesEqual(path, _testNotificationPath)) {
+                _testNotificationPath.clear();
+            }
+            if (pathIsIgnored(path)) {
+                paths.remove(path);
```

Comment:
Is this removal safe? In the STL, AFAIK, it would not work reliably unless you use iterators.

Quoting from https://doc.qt.io/qt-6/qset-iterator.html:

> [QSet](https://doc.qt.io/qt-6/qset.html)<T>::iterator allows you to iterate over a [QSet](https://doc.qt.io/qt-6/qset.html) and to remove items (using [QSet::erase](https://doc.qt.io/qt-6/qset.html#erase)()) while you iterate.

Reply:
thx